src/optimizer/query_graph.c

```diff
@@ -1305,6 +1306,14 @@ qo_add_node (PT_NODE * entity, QO_ENV * env)
 	      if (xasl)
 		{
 		  QO_NODE_NCARD (node) = (unsigned long) xasl->cardinality;
+
+		  /* check limit value */
+		  limit_val = pt_get_query_limit (env->parser, entity->info.spec.derived_table);
+		  if (limit_val > 0 && (unsigned long) limit_val < QO_NODE_NCARD (node))
+		    {
+		      QO_NODE_NCARD (node) = (unsigned long) limit_val;
```

Comment:
When limit is used with offset, card becomes the sum of row_count and offset.
```
select /*+ recompile */ count(*) 
from (select /*+ no_merge */ * from t111 where col1 is not null limit 20000 offset 10000) a, 
        (select /*+ no_merge */ * from t111 where col3 = 1) b,
        t111 c 
where c.col3 = 1 
   and a.col1 = b.col1 
   and a.col1 = c.col1
   and b.col1 = c.col1;
```
```
Query plan:

nl-join (inner join)
    edge:  term[0] AND term[1]
    outer: idx-join (inner join)
               outer: sscan
                          class: b node[1]
                          cost:  76 card 10000
               inner: iscan
                          class: c node[2]
                          index: idx1 term[2] AND term[3] (covers)
                          cost:  3 card 10000
               cost:  5103 card 1000
    inner: sscan
               class: a node[0]
               sargs: term[0] AND term[1]
               cost:  228 card 30000
    cost:  248403 card 1
```
Please refer to tastecases for CBRD-25080.txt.